app/routes/surveys/EditSurveyRoute.js

```diff
@@ -1,43 +1,82 @@
 import { connect } from 'react-redux';
 import prepare from 'app/utils/prepare';
 import { compose } from 'redux';
-import { editSurvey, fetch, deleteSurvey } from '../../actions/SurveyActions';
+import {
+  editSurvey,
+  fetch,
+  deleteSurvey,
+  fetchTemplate
+} from '../../actions/SurveyActions';
 import SurveyEditor from './components/SurveyEditor/SurveyEditor';
 import { LoginPage } from 'app/components/LoginForm';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
-import { selectSurveyById } from 'app/reducers/surveys';
+import { selectSurveyById, selectSurveyTemplate } from 'app/reducers/surveys';
 import { push } from 'react-router-redux';
 import loadingIndicator from 'app/utils/loadingIndicator';
 
+const loadData = (props, dispatch) => {
+  const { surveyId } = props.params;
+  const { templateType } = props.location.query;
+  if (templateType) {
+    return Promise.all([
+      dispatch(fetchTemplate(templateType)),
+      dispatch(fetch(surveyId))
+    ]);
+  }
+  return dispatch(fetch(surveyId));
+};
+
 const mapStateToProps = (state, props) => {
+  const notFetching = !state.surveys.fetching;
   const surveyId = Number(props.params.surveyId);
   const survey = selectSurveyById(state, { surveyId });
+  const templateType = props.location.query.templateType;
+  const template = selectSurveyTemplate(state, { ...props, templateType });
+
+  const initialEvent = survey.event && {
+    value: survey.event.id,
+    label: survey.event.title
+  };
+
+  let initialValues = null;
+  if (notFetching && !(templateType && !template)) {
```

Comment:
I tried having just `notFetching` here, but it really seemed unpredictable and sporadic. It _usually_ does the trick, but I can't seem to see the pattern to when it decides to say it's done fetching despite `template` not being available in state yet. Could be a problem somewhere else in the stack but I couldn't find it.